multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/handlers/ResourceBatchCalculator.java

```diff
@@ -0,0 +1,56 @@
+package org.cloudfoundry.multiapps.mta.handlers;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.cloudfoundry.multiapps.mta.handlers.v3.DescriptorHandler;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.cloudfoundry.multiapps.mta.model.WeightedResource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ResourceBatchCalculator {
+
+    private final DeploymentDescriptor deploymentDescriptor;
+    private final DescriptorHandler descriptorHandler;
+
+    public ResourceBatchCalculator(DeploymentDescriptor deploymentDescriptor) {
+        this.deploymentDescriptor = deploymentDescriptor;
+        this.descriptorHandler = new DescriptorHandler();
+
+    }
+
+    public Map<Integer, List<Resource>> groupResourcesByWeight(List<Resource> resources) {
+        List<WeightedResource> result = new ArrayList<>(20);
+        for (Resource resource : resources) {
+            int weight = findHeaviestDependencyWeight(resource, 0, deploymentDescriptor);
+            result.add(new WeightedResource(resource, weight));
+        }
+        return convertToMap(result);
+    }
+
+    private Map<Integer, List<Resource>> convertToMap(List<WeightedResource> resourceSet) {
```

Comment:
`convertToMap` isn't really descriptive of what this does.
Why are you converting to a map? What kind of map is it? Why do you need a map?
A better name would be `groupByWeight`